File: backend/app/schemas/document.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class UploadResponse(BaseModel):
    document_id: UUID
    status: str


class DocumentListItem(BaseModel):
    id: UUID
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    total_chunks: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class DocumentDetail(DocumentListItem):
    file_path: str
    extracted_text: str


class ChunkRead(BaseModel):
    id: UUID
    document_id: UUID
    chunk_index: int
    chunk_text: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/document.ts
export interface DocumentListItem {
  id: string;
  filename: string;
  original_filename: string;
  file_type: string;
  file_size: number;
  total_chunks: number;
  created_at: string;
}

export interface DocumentDetail extends DocumentListItem {
  file_path: string;
  extracted_text: string;
}

export interface DocumentChunk {
  id: string;
  document_id: string;
  chunk_index: number;
  chunk_text: string;
  created_at: string;
}

export interface UploadResponse {
  document_id: string;
  status: string;
}
